weld/llvm.rs

```diff
@@ -765,223 +781,279 @@ impl LlvmGenerator {
         Ok(())
     }
 
+    /*********************************************************************************************
+    //
+    //
+    //
+    // TODO(shoumik): Refactor code above this point!
+    //
+    //
+    //
+    ********************************************************************************************/
+
+    /*********************************************************************************************
+    //
+    // Utilities
+    //
+    *********************************************************************************************/
+
+    /// Returns the Type, LLVM type (as a string), and LLVM symbol name (as a string).
+    fn symbol_info<'a>(&mut self, func: &'a SirFunction, sym: &Symbol) -> WeldResult<(&'a Type, String, String)> {
+        let ty = func.symbol_type(sym)?;
+        let llvm_ty = self.llvm_type(ty)?;
+        let llvm_str = llvm_symbol(sym);
+        Ok((ty, llvm_ty, llvm_str))
+    }
```

Comment:
Or if we *always* get the last two fields and only sometimes get the first one, maybe just make this return those two fields, and call `symbol_type` when you need the other one. Then you should also pick a more descriptive name than `symbol_info`, maybe `llvm_name_and_type` or `llvm_info` or something.